src/registers/debug.rs

```diff
@@ -0,0 +1,603 @@
+        }
+    }
+
+    const fn bit_position(n: DebugAddressRegisterNumber) -> u8 {
+        18 + 4 * n.0
+    }
+}
+
+impl TryFrom<usize> for HwBreakpointSize {
+    type Error = TryFromIntError;
+
+    fn try_from(size: usize) -> Result<Self, Self::Error> {
+        Self::new(size).ok_or(TryFromIntError(()))
+    }
+}
+
+/// A valid value of the [`Dr7`] debug register.
+///
+/// In addition to the [`Dr7Flags`] this value has a condition field and a size field for each debug address register.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct Dr7Value {
+    bits: u64,
+}
+
+impl From<Dr7Flags> for Dr7Value {
+    fn from(dr7_flags: Dr7Flags) -> Self {
+        Self::from_bits_truncate(dr7_flags.bits())
+    }
+}
+
+impl Dr7Value {
+    const fn mask() -> u64 {
+        let field_mask = (1 << 32) - (1 << 16);
+        let flag_mask = Dr7Flags::all().bits();
+        field_mask | flag_mask
+    }
+
+    /// Convert from underlying bit representation, unless that representation contains bits that do not correspond to a field.
+    #[inline]
+    pub const fn from_bits(bits: u64) -> Option<Self> {
+        if (bits & Self::mask()) == 0 {
+            Some(Self { bits })
+        } else {
+            None
+        }
+    }
+
+    /// Convert from underlying bit representation, dropping any bits that do not correspond to fields.
+    #[inline]
+    pub const fn from_bits_truncate(bits: u64) -> Self {
+        Self {
+            bits: bits & !Self::mask(),
+        }
+    }
+
+    /// Convert from underlying bit representation, preserving all bits (even those not corresponding to a defined field).
+    ///
+    /// # Safety
+    ///
+    /// The bit representation must be a valid [`Dr7Value`].
+    #[inline]
+    pub const unsafe fn from_bits_unchecked(bits: u64) -> Self {
+        Self { bits }
+    }
+
+    /// Returns the raw value of the fields currently stored.
+    #[inline]
+    pub const fn bits(&self) -> u64 {
+        self.bits
+    }
+
+    /// Returns the [`Dr7Flags`] in this value.
+    #[inline]
+    pub const fn flags(self) -> Dr7Flags {
+        // SAFETY: The extra bits used in fields are valid in our representation of `Dr7Flags`.
+        unsafe { Dr7Flags::from_bits_unchecked(self.bits) }
+    }
+
+    /// Returns a mutable reference to the [`Dr7Flags`] in this value.
+    ///
+    /// This can be used to modify the flags of this value in-place.
+    #[inline]
+    pub fn flags_mut(&mut self) -> &mut Dr7Flags {
+        // SAFETY: The extra bits used in fields are valid in our representation of `Dr7Flags`.
+        // SAFETY: `Dr7Flags` has the same representation as `Dr7Value`.
+        unsafe { mem::transmute(self) }
+    }
+
+    /// Returns the condition field of a debug address register.
+    pub fn get_condition(&self, n: DebugAddressRegisterNumber) -> HwBreakpointCondition {
+        let pos = HwBreakpointCondition::bit_position(n);
+        let condition = (self.bits >> pos) & 0b11;
```

Comment:
You are handling bits with `<<`, `>>`, `&`, and `|`, but you can also use `get_bit`, `get_bits`, `set_bit`, and `set_bits` with `use bit_field::BitField;`. I think it is easier to see what is happening.
```suggestion
        let condition = self.bits.get_bits(pos..pos + 2);
```

Reply:
Thanks for the hint! I adjusted `(get,set)_(condition,size)` accordingly.